code/ui/manager/src/components/preview/FramesRenderer.tsx

```diff
@@ -57,50 +65,42 @@ export const FramesRenderer: FC<FramesRendererProps> = ({
   });
   const active = getActive(refId, refs);
 
-  const styles = useMemo<CSSObject>(() => {
-    // add #root to make the selector high enough in specificity
-    return {
-      '#root [data-is-storybook="false"]': {
-        display: 'none',
-      },
-      '#root [data-is-storybook="true"]': {
-        display: 'block',
-      },
-    };
-  }, []);
+  const autoInjectedRefs = Object.values(refs).filter((ref) => {
+    return ref.type === 'auto-inject';
+  }, {});
 
+  const selectedRef = refs[refId];
   const [frames, setFrames] = useState<Record<string, string>>({
     'storybook-preview-iframe': getStoryHref(baseUrl, storyId, {
       ...queryParams,
       ...(version && { version }),
       viewMode,
     }),
+    ...(selectedRef
+      ? {
+          [`storybook-ref-${selectedRef.id}`]: `${selectedRef.url}/iframe.html?id=${storyId}&viewMode=${viewMode}&refId=${selectedRef.id}${stringifiedQueryParams}`,
+        }
+      : {}),
+    ...autoInjectedRefs.reduce((acc, ref) => {
+      return {
+        ...acc,
+        [`storybook-ref-${ref.id}`]: `${ref.url}/iframe.html?id=${storyId}&viewMode=${viewMode}&refId=${ref.id}${stringifiedQueryParams}`,
+      };
+    }, {}),
```

Comment:
Can we combine this to one array to avoid duplication? `[...autoInjectedRefs, selectedRef]`